README.md

```diff
@@ -1,86 +1,108 @@
 
-![compute-comparison](./img/compute-comparison.png)
+In a few months YAP goes viral and they start getting users from across the world. They find that photos search requests are have regular patterns based on geographic location. They want to keep their end user latency to be less than 1.5 secs and setup CDNs for major locations.
 
-The idea is that runtime system should choose the specific implementation based on user defined constraints.
+YAP has now become a popular global search service and consistently gets 100,000 reqs/s. YAP now decides to use dedicated hardware to serve consistent traffic efficiently. Bursty traffic can still be handled through autoscaling and setting appropriate thresholds.
 
-## Automated least priviledge security [^4]
+In YAP's journey we see business requirements and user behaviour drive architectural changes. The application logic constantly has to adapt to integrate with architectural components that are best suited to the situation as YAP usage grows.
 
-A common theme across cloud provider is ensuring least priviledge access to various entities. This is done by giving permissions for various APIs to a role and then attaching this role to a user or compute instance which makes the API calls.
+### Idea
 
-In this context least priviledge means giving an entity only the required permissions for it to perform it's actions. Currently this is process of deciding required permissions is completely manual and extremely tedious and error prone.
+A moden programming system should understand changing requirements from high level constraints and choose appropriate components. Here's a table mapping the relationship in the above table.
 
-The main reasoning behind this is that erroneous or malicious changes outside the scope of the application are rendered impossible. Since most cloud providers give access to runtime systems that can run arbitrary code. A system with extra priviledges can potentially be used as an avenue for attacking the other parts of the organizations IT infrastructure.
+| Situation | Component |
+| -- | -- |
+| 1000 - 3000 req/s | stateless functions |
+| 10,000 - 30,000 req/s | servers running in containers |
+| 1.5 secs end user latency | CDN |
+| 100,000 req/s | servers running on dedicated VMs with thresholds and autoscaling |
 
-A better cloud declaration language/SDK would automatically be able to generate the required permissions based by analyzing the API calls in the application code.
+While this is a straight forward real world components modelling will consider numerous properties including cost, throughput, latency, integrations, operational overhead. A programming system with this information will be able to derive the architectural components from the application annotated by constraints.
```

Comment:
Yeah we discussed that some of the concerns around these were on the human side. Such as over reliance on the tool, developing a lax attitude towards security/optimization because tooling is supposed to handle most of it.

I'll figure out a way to include this aspect in the next draft. Collect this in the issue.